### ISSUE TITLE
#error -- unsupported ICC configuration! Only ICC 15.0 and ICC 16.0 on Linux x86_64 are supported!

### ISSUE THREAD

**Issue by pedronahum** [intel/torch#7]
Hi,

icc (ICC) 17.0.4 20170411 fails to compile the code. 

It looks like there are some cuda components (eg, host_config.h and cuda_runtime.h) that only allow ICC15 and 16. 

Is this expected?

Thanks,

Pedro N.

**Comment by MlWoo** [comment]
@pedronahum 
Yes, The cuda component will check the version of ICC. You can modify the cuda files to give the green light to this step if you insist to use ICC20170411 to compile it.

**Comment by pedronahum** [comment]
@MlWoo
 
Many thanks for getting back to me. I did the changes, yet it seems that some flags might missing. In particular, I get many errors like

error: expected an expression
    for (int b = 0; b < batch_size; b++) {
         ^
The error arrives while building the C object lib/THNN/CMakeFiles/THNN.dir/init.c.o

Could it be that the "-std=c99" flag is missing in this step?

Please note that code compiles with gcc

**Comment by MlWoo** [comment]
@pedronahum 
Yes, it maybe depends on your cmake version perhaps, you can refer to [cmakefile ](https://github.com/MlWoo/nn/blob/intel/lib/THNN/CMakeLists.txt#L26), you can add the SET(CMAKE_C_FLAGS "-std=c99 ${CMAKE_C_FLAGS}") outside of the if statements to help you skip the error, I will fix the bug in future.

**Comment by pedronahum** [comment]
Hi @MlWoo 

Indeed, my cmake version (3.6.3) seems to be playing tricks with the installation. After adding the above logic to the file, I now get the following error:

In file included from /usr/include/c++/4.9/mutex(35),
                 from /media/pedro/hdd/intel-ai/torch/extra/cutorch/lib/THC/THCStream.cpp(3):
/usr/include/c++/4.9/bits/c++0x_warning.h(32): error: #error directive: This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
  #error This file requires compiler and library support for the \
   ^

What would you recommend?
 
Thanks,

Pedro N.

**Comment by MlWoo** [comment]
@pedronahum 
Yeah, you should add CMAKE_CXX_STANDARD to the [file](https://github.com/torch/cutorch/blob/92e9c08ca008bd5c2246ac028d6174f42429874d/rocks/cutorch-scm-1.rockspec), or maybe perhaps the [file](https://github.com/torch/cutorch/blob/92e9c08ca008bd5c2246ac028d6174f42429874d/rocks/cutorch-1.0-0.rockspec) and so do in the cudnn folder. If you are not familiar to CMake, it is a good idea to add export CMAKE_CXX_STANDARD=c++11 or gnu++11 to your environment.
